src/stateless/slider.vue

```diff
@@ -0,0 +1,545 @@
+                    </div>
+
+                    <div class="bar__rail">
+                        <div class="rail rail--passive"></div>
+                        <div :class="{'rail--exceeds': limit && value > limitValue}" :style="{width: `${position}px`}" class="rail rail--active"></div>
+                        <div :style="{left: `${position}px`}" class="rail__handle">
+                            <div class="rail__handle-dot"></div>
+                        </div>
+                    </div>
+                </div>
+
+                <div class="slider-helper-text"></div>
+            </div>
+        </div>
+    </div>
+</template>
+
+<script>
+import Input from './input.vue'
+
+export default {
+    components: {
+        inputElement: Input,
+    },
+    props: {
+        min: { type: Number, required: true },
+        max: { type: Number, required: true },
+        limit: { type: Number, required: false },
+        minLabel: { type: String, required: false },
+        maxLabel: { type: String, required: false },
+        label: { type: String },
+        step: { type: Number },
+        value: { type: Number },
+        unit: { type: String, required: false },
+        disabled: { type: Boolean, default: false },
+        disabledText: { type: String, required: false },
+        theme: { type: String, default: 'dark' },
+        size: { type: String, default: 'normal' },
+        alignment: { type: String, default: 'left' },
+        isValid: { type: Function, required: false },
+    },
+    data () {
+        return {
+            isDomReady: false,
+            isChanged: false,
+            isDragging: false,
+        }
+    },
+    computed: {
+        limitValue () {
+            return this.limit || this.max
+        },
+        minLabelValue () {
+            return this.minLabel || this.min.toString()
+        },
+        maxLabelValue () {
+            return this.maxLabel || this.limitValue.toString()
+        },
+        index () {
+            return (this.value - this.min) / this.step
+        },
+        stepsCount () {
+            return (this.limitValue - this.min) / this.step
+        },
+        stepPercentage () {
+            return 1 / this.stepsCount
+        },
+        percentage () {
+            return (this.value - this.min) / (this.limitValue - this.min)
+        },
+        position () {
+            return this.isDomReady ? Math.max(0, Math.min(this.index, this.stepsCount)) * this.stepPercentage * this.sliderWidth : 0
+        },
+        sliderWidth () {
+            return this.isDomReady ? this.$refs.bar.clientWidth : 0
+        },
+        sliderOffset () {
+            return this.isDomReady ? this.$refs.bar.getBoundingClientRect().x : 0
+        },
+        decimalPlacesCount () {
+            let decimals = this.step.toString().split('.')[1]
+            return decimals ? decimals.length : 0
+        },
+        inputType () {
```

Comment:
[minor] I would inline this in template.

Reply:
It's easier to write tests this way.

Reply:
Maybe change it to `isWholeNumber` which just returns a boolean (`this.decimalPlacesCount === 0`) and can easily be tested. `'number'` and `'float'` values are very specific to the input component and I would rather inline them.